src/client/common/extensions.ts

```
@@ -65,3 +66,35 @@ String.prototype.fileToCommandArgument = function (this: string): string {
     }
     return this.toCommandArgument().replace(/\\/g, '/');
 };
+
+// tslint:disable-next-line:interface-name
+declare interface Promise<T> {
+    /**
+     * Explicitly states that promise will run asynchronously.
+     */
+    doNotWait(onComplete?: () => void, onError?: (reason?: {}) => void): void;
+}
+
+/**
+ * Explicitly tells that promise should be run asynchonously.
+ */
+Promise.prototype.doNotWait = function <T>(this: Promise<T>, onComplete?: () => void, onError?: (reason?: {}) => void) {
+    this
+        .catch(reason => {
+            if (onError) {
+                //vscode.window.createOutputChannel(STANDARD_OUTPUT_CHANNEL).appendLine(`Promise exception: ${reason.ToString()}`);
+                onError(reason);
+            }
+        })
+        .then(() => {
+            if (onComplete) {
+                onComplete();
+            }
+        })
+        .catch(() => {
+            if (onError) {
+                //vscode.window.createOutputChannel(STANDARD_OUTPUT_CHANNEL).appendLine(`Promise exception: ${reason.ToString()}`);
+                onError();
```

Shouldn't we be capturing the reason and passing that through to the `onError` function

---

Which logger should I use

---

What's the purpose of `doNotWait`. If i'm not mistaken, we'd only use this in case where we don't care about any exceptions raised at all. Hence I'd say, we don't care about the errors, that's why we didn't have any error handlers in the first place. If we do want them, then i'd put the logging in the code itself, rather than creating an extension method.

So, all we need is:
```typescript
/**
 * Explicitly tells that promise should be run asynchonously.
 */
Promise.prototype.doNotWait = function(this: Promise) {
    this.catch(()=>{});
    // or     this.catch(nop);
};
```

Now, when you look at the code, this extension feels kind of redundant or we should rename it to something such as `ignoreErrors()`.

Because technically this still returns a promise that will get resolved. So it not about waiting or not. We just want to ignore the errors.

---

The purpose is to explicitly state that we don't want to await here by design. Generally caller doesn't care about exceptions, but may appreciate if they are logged.

---

In C# we have `SilenceException` and `DoNotWait`. The latter indicates that task runs freely and is not awaited, but it does rethrow exceptions.

---

Let's make this one into `ignoreErrors`. We can discuss rethrowing separately